### ISSUE TITLE
iOS 17: Feature requests, please add in comments

### ISSUE THREAD

**Issue by shaps80** [shaps80/SwiftUIBackports#48]
In order to best serve the community, please request features (or react to an existing one) using the comments on this issue. I will then prioritise all efforts to try and fill those backports where possible 👍

> For Swift (non-UI) requests please see [SwiftBackports](https://github.com/shaps80/SwiftBackports/issues/2)

–––

I'm surprised I need to say this, but please **do not** add comments below that are unrelated to this ticket!

**Comment by ShikiSuen** [comment]
Generic parameter 'Value' shadows generic parameter from outer scope with the same name; this is an error in Swift 6.
https://github.com/shaps80/SwiftUIBackports/blob/8e92217f52f5d3981a96870cc518a095a13b3e08/Sources/SwiftUIBackports/Internal/Environment%2BString.swift#L93-L96
<img width="1404" alt="image" src="https://github.com/shaps80/SwiftUIBackports/assets/3164826/9df59cf8-6fd4-49f4-95cc-ee4ef164b0bd">

What about rewriting it as follows?:
```swift
  final class Store<StoredValue>: ObservableObject {
    var value: StoredValue?
  }
```

**Comment by shaps80** [comment]
I wonder why I didn't see this warning before, thanks!

Also, yeah the optional set to nil used to be required when this was first written, but you're right its no longer required 👍

**Comment by shaps80** [comment]
- Section expand/collapse
- KeyFrameAnimator

**Comment by yujinqiu** [comment]
backport `symbolEffect`  will be appreciated

**Comment by shaps80** [comment]
> backport `symbolEffect` will be appreciated

I'm not sure this one is possible, but I may investigate to confirm 👍

**Comment by yujinqiu** [comment]
`ContentUnavailableView`  is also useful and it looks not hard to backport.

**Comment by PangMo5** [comment]
[scrollTarget(isEnabled:)](https://developer.apple.com/documentation/swiftui/view/scrolltarget(isenabled:)), [scrollTargetBehavior(_:)](https://developer.apple.com/documentation/swiftui/view/scrolltargetbehavior(_:)), [scrollTargetLayout(isEnabled:)](https://developer.apple.com/documentation/swiftui/view/scrolltargetlayout(isenabled:))
It would be really useful if those methods were backported.

**Comment by shaps80** [comment]
@PangMo5 I agree! I'll need to dig into these in details, I did take a very quick look a while ago and thought perhaps this couldn't be done, but I'll need to get into the weeds before I know for sure 👍

**Comment by BastianKusserow** [comment]
A backport of `.contentTransition(.numericText(value:))` would be awesome!

**Comment by shaps80** [comment]
> A backport of `.contentTransition(.numericText(value:))` would be awesome!

I considered this actually, I don't think it's that difficult to implement actually, could be done essentially with an `AnimatableModifier` I think – however I'd rather backport the entire `contentTransition` API vs just a specific version of it, and tbh not all of them are so easy AFAICT.

I'll consider it though 👍

**Comment by BastianKusserow** [comment]
@shaps80 would be interested on how you would implement the numeric text transition, as I've struggled with this quite a lot when I tried it. I thought one could reuse the numericText(countDown:) transition but I was not sure how I get the old and the new value for comparison..

**Comment by shaps80** [comment]
> @shaps80 would be interested on how you would implement the numeric text transition, as I've struggled with this quite a lot when I tried it. I thought one could reuse the numericText(countDown:) transition but I was not sure how I get the old and the new value for comparison..

Definitely dig into animatable modifier, it's not a quick response that I can leave here. And I just don't have the time right now to Explore. Sorry

**Comment by bstillitano** [comment]
`searchScopes` would be great to have back ported to at least iOS 15

**Comment by shaps80** [comment]
@bstillitano good shout! I did actually start (unfinished) a searchable modifier backport when it was first released. I can't remember why I didn't finish it. I could try completing that and adding scopes as well as tokens actually cause that's a great API.

**Comment by vmanot** [comment]
@shaps80 would love to know if backporting [`Shader`](https://developer.apple.com/documentation/swiftui/shader) is even possible.

**Comment by shaps80** [comment]
@vmanot that's def out of my scope tbh, but I'd be surprised if it wasn't possible at least 1 or 2 versions back. I'd definitely love a PR exploring this!

**Comment by vmanot** [comment]
@shaps80 glad to know there are no obvious reasons why not! And yes, will definitely PR if I find the bandwidth to push on this front.

**Comment by jadar** [comment]
@shaps80 I'd be willing to take a crack at that searchable modifier backport, if you wanted to hand it off. :)

**Comment by shaps80** [comment]
> @shaps80 I'd be willing to take a crack at that searchable modifier backport, if you wanted to hand it off. :)

Sure! I have a start somewhere. Let me try and find it and see if I can drop a starting branch off. 

If not I'll let you know and you can start off from scratch. Gimme a couple days.

**Comment by hstdt** [comment]
how about backport `.buttonRepeatBehavior(.enabled)`

**Comment by nlash-livly** [comment]
would love to have `.scrollPosition(id:)`

**Comment by futuretap** [comment]
would love to have `UIViewRepresentable.sizeThatFits(_:uiView:context:)` and `View.contentMargins(_:for:)`.

**Comment by fl034** [comment]
[TipKit](https://developer.apple.com/documentation/tipkit) would be awesome!

**Comment by shaps80** [comment]
@fl034 I actually started a fairly full on backport of TipKit, but then I realised some issues. We'd equally need to backport the new `Predicate` APIs (including the macros) since its syntax heavily relies on that.

It's totally possible, especially since `Predicate` is actually open sourced through the Foundation efforts on Apple's repo. However it's a huge effort and I've simply not had the time to properly get into it.

I also, love TipKit and agree it'd be a great addition.

**Comment by kolyneh** [comment]
Hi Shaps, thanks for your good work! We would love to have `.presentationCompactAdaptation(_ adaptation:)` and `presentationCompactAdaptation(horizontal:, vertical: )`.
